### PR TITLE
CONTRIBUTING.md: add Github contribution guidelines

### PR DESCRIPTION
**Pull request by jsquyres**
We have a small number of requirements for contributions (e.g.,
"Signed-off-by"), so let's make sure that people have an easy way of
knowing these things.

Signed-off-by: Jeff Squyres <jsquyres@cisco.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+contributions are compatible with that license.
+
+To that end, we require that all Git commits contributed to Open MPI
+have a "Signed-off-by" token indicating that the commit author agrees
+with [Open MPI's Contributor's
+Declaration](https://github.com/open-mpi/ompi/wiki/Admistrative-rules#contributors-declaration).
+
+If you have not already done so, please ensure that:
+
+1. Every commit contains exactly this token.  You can add this token
```
Comment (jjhursey) [inline CONTRIBUTING.md]: Change `this token` to `the "Signed-off-by" token`

Comment (jsquyres) [reply CONTRIBUTING.md]: Done!

**Comment by rhc54** [comment]
@jsquyres Something didn't work right - I'm not seeing the contribution file link when opening an issue or PR.

**Comment by jjhursey** [comment]
Bonk - look in `.github/CONTRIBUTING.md`

**Comment by jsquyres** [comment]
Just to follow up on this PR: it *is* working.  But Github has lame-ified the link.  😦 

github-contributing-link

**Pull request merged**